### PR TITLE
fix: trace no items

### PR DESCRIPTION
**Pull request by jnoob**
finish is called without any item

### CODE REVIEW

File: dataloader.go [hunk 1]
```diff
@@ -414,7 +414,9 @@ func (b *batcher) batch(originalContext context.Context) {
 	}
 
 	ctx, finish := b.tracer.TraceBatch(originalContext, keys)
-	defer finish(items)
+	defer func() {
```
Comment (vivek-ng) [inline dataloader.go:417]: What difference does this make and what exactly are you trying to solve ?
How is `defer finish(items)` any different than what you have ?

Comment (jnoob) [reply dataloader.go:417]: With `defer finish(items)`, the function will always receive a slice without any element

Comment (jnoob) [reply dataloader.go:417]: `defer finish(items)`  -- `items` is evaluated when `finish` is deferred, so it's always a empty slice
`defer func(){ finish(items) }()` -- `items` is evaluated when the deferred func is executed, so the subsequent changes will be applied.

Comment (vivek-ng) [reply dataloader.go:417]: Ah , makes sense 👍 . Can you please write a unit test for this ? You can probably mock `tracer` as it's an interface and provide your own definition for finish.

**Comment by vivek-ng** [comment]
@tonyghita can you please review this ? I think this is an important bug which needs to be fixed as it breaks tracing .

File: dataloader.go (hunk 1 above)
Comment (jnoob) [reply dataloader.go:417]: I am not good at writing unit tests, I'd be glad to receive your comments

**Pull request merged**